Standardise file names and location

javajawa opened the pull request:
Change bpkg-build to output files in the package's directory using the standard `{PACKAGE}_{VERSION}_{ARCHITECTURE}.deb` filename format.

This introduces a check that the `Architecture:` of a package is defined in either the `control` file, or in the `ARCHITECTURE` environment variable. If it is supplied in both, they must match.

javajawa committed:
bpkg-build: Place built files in package directory

The use of POOLDIR and the bpkg-make-list structure within bpkg-build
creates a unintuitive coupling. Instead, debs will be placed in the
build directory.

javajawa committed:
bpkg-build: Allow an optional "destination" parameter

The new third parameter, destination, takes a path in which the build
.deb file and associated .deb.dat file should be placed.
If defaults to the directory of the package being built. It will be
created if it does not exist.

javajawa committed:
bpkg-build: Add some clarification and a usage message

javajawa committed:
bpkg-build: Add a basic understanding of architecture

The change requires that either an architecture be supplied in the
environment line or in the control file for a package.
If it is in the environment, then the Makefile must be configured
to produce a correct build/manifest result for that target environment.

ethulhu commented on src/bpkg-build/bpkg-build:
```diff
@@ -50,13 +50,18 @@ then
+# Finalise the determined version
 VERSION=${VERSION:-$2}
 
+# Determins where the file is to be written to
```
s/Determins/Determines/

javajawa replied:
cba to reopen ssh, gonna make github do the work

```suggestion
# Determines where the file is to be written to
```

ethulhu reviewed (approved):
LGTM, ship it.

javajawa committed:
bpkg-build: bpkg-build is now configurationless.

With the removal of the logic to pull into the pool directory,
there is no longer any external configuration required for bpkg-build.
As such, the code to pull in /etc/bpkg.conf has been removed.

javajawa commented on src/bpkg-build/bpkg-build:
```diff
@@ -23,24 +19,47 @@ check xz
 NAME=${NAME#Package: }
 
+# The architecture is specified in the control file,
+# or supplied at build time for multi-arch pacakges.
```
```suggestion
# or supplied at build time for multi-arch packages.
```

javajawa committed:
Fix typos from code review
